octreelib/grid/grid.py

```diff
@@ -107,23 +108,118 @@ def insert_points(self, pose_number: int, points: PointCloud):
             self.__pose_voxel_coordinates[pose_number].append(target_voxel)
             self.__octrees[target_voxel].insert_points(pose_number, voxel_points)
 
-    def map_leaf_points(self, function: Callable[[PointCloud], PointCloud]):
+    def map_leaf_points(
+        self,
+        function: Callable[[PointCloud], PointCloud],
+        pose_numbers: Optional[List[int]] = None,
+    ):
         """
         Transform point cloud in each node of each octree using the function
         :param function: Transformation function PointCloud -> PointCloud. It is applied to each leaf node.
+        :param pose_numbers: List of pose numbers to map.
         """
         for voxel_coordinates in self.__octrees:
-            self.__octrees[voxel_coordinates].map_leaf_points(function)
+            self.__octrees[voxel_coordinates].map_leaf_points(function, pose_numbers)
 
-    def get_leaf_points(self, pose_number: int) -> List[Voxel]:
+    def map_leaf_points_cuda_ransac(
+        self,
+        n_poses_per_batch: int = 1,
+        threshold: float = 0.01,
+        n_hypotheses: int = 1024,
```

Comment:
hypotheses_number?

Reply:
Let's avoid letter variable naming

Reply:
Not sure because there are already things in the library which are named "n_*something*": `n_points`, `n_leaves`, `n_nodes`, which are methods of `Grid`, `Octree` and `OctreeManager`. We can change that if necessary or keep with the convention.

Reply:
That's properties naming. I'm not sure if this also correct, but letter naming is terrible, let's fix this at least.

Reply:
done